### PR TITLE
[dotnet] Parse the -gcc_flags and -link_flags mtouch/mmp arguments and pass them to the native linker.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Review by chamons — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ParseBundlerArgumentsTaskBase.cs [hunk 1]
```diff
@@ -55,6 +61,7 @@ public override bool Execute ()
 			if (!string.IsNullOrEmpty (ExtraArgs)) {
 				var args = CommandLineArgumentBuilder.Parse (ExtraArgs);
 				List<string> xml = null;
+				List<string> custom_link_flags = null;
```
Comment (mandel-macaque) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ParseBundlerArgumentsTaskBase.cs:64]: Why do we use snake_case here and camelCase in the other vars?

**Review by mandel-macaque — APPROVED** [review]
Small style comment.

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* link sdk/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2494
* MTouch tests/NUnit: https://github.com/xamarin/xamarin-macios/issues/13517

**Review by emaf — APPROVED** [review]

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ParseBundlerArgumentsTaskBase.cs (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ParseBundlerArgumentsTaskBase.cs]: Fixed!

**Review by mauroa — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated:

* interdependent-binding-projects/Mac Catalyst [dotnet]/Debug [dotnet]: https://github.com/xamarin/maccore/issues/2414
* MTouch tests/NUnit: https://github.com/xamarin/xamarin-macios/issues/13517

**Pull request merged**